### PR TITLE
Fix hinted-mods.py for compatibility with Nuitka 0.6.11

### PR DESCRIPTION
**Pull request by jackz314**

### CODE REVIEW

File: hinted-compilation/hinted-mods.py [hunk 1]
```diff
@@ -86,7 +86,7 @@ def get_checklist(full_name):
     """
     if not full_name:  # guard against nonsense
         return []
-    mtab = full_name.split(".")  # separate components by dots
+    mtab = str(full_name).split(".")  # separate components by dots
```
Comment (kayhayen) [inline hinted-compilation/hinted-mods.py]: I think you want to use `splitPackageName` in a loop here, until the package name becomes `None`, that avoids the string operation.

File: hinted-compilation/hinted-mods.py [hunk 2]
```diff
@@ -322,20 +322,20 @@ def onModuleEncounter(self, module_filename, module_name, module_kind):
             Example: (False, "because it is not called").
         """
         full_name = module_name
-        elements = full_name.split(".")
+        element = full_name.getTopLevelPackageName()
```
Comment (kayhayen) [inline hinted-compilation/hinted-mods.py]: The `top_level_package_name` would be way better than `element` in downwards code to understand things, `element` is just meaningless.

**Review by kayhayen — CHANGES_REQUESTED** [review]

File: hinted-compilation/hinted-mods.py (hunk 1 above)
Comment (jackz314) [reply hinted-compilation/hinted-mods.py]: Must've missed that when reading the docs, I'll fix that.

**Comment by kayhayen** [comment]
I changed the title, because I think it makes it work with 0.6.11

I am going to add tests as a workflow for this repo eventually, so we can know  if it's broken currently or not.

**Comment by h4ckzard** [comment]
Broken. Now it drops non-direct dependencies that are called inside sub-modules.
In fact, all incompatibility was solved by a `full_name = str(module_name)` in line 324 instead all of this commit (yes, its a crutch)

**Comment by kayhayen** [comment]
@h4ckzard can you make a PR?

**Pull request merged**